cocotb/handle.py

```diff
@@ -698,6 +698,11 @@ def _set_value(self, value, call_sim):
         Raises:
             TypeError: If target is not wide enough or has an unsupported type
                  for value assignment.
+
+        .. deprecated:: 1.5
+            :class:`dict` objects are no longer accepted as values for assignment.
+            Convert the dictionary to an integer before assignment using
+            ``sum(v << (d['bits'] * i) for i, v in enumerate(d['values']))``.
```

Comment:
This implementation isn't the same as the one we are deprecating, it reverses the entries!

Reply:
That's due to how it's building the integer. I got tricked by using `reversed` at first. The test shows it's correct. But I will do what you had mentioned above.